### DISCUSSION TITLE
Are there plans to publish rspack to crates.io?

### DISCUSSION THREAD

**Discussion by ssccchh** [web-infra-dev/rspack#2195] [General]
I want to use rspack as part of my build toolchain, but currently I can only make build scripts in the rspack project.

![image](https://user-images.githubusercontent.com/26810503/224590789-fd1051a4-db93-4f0b-9b9c-2a2c63a0b6fd.png)

In crate.io, I did not find the crate corresponding to rspack

**Comment by hardfist** [comment]
no plan to publish rspack crate yet, but you could still use rspack by github repo url

**Reply by ssccchh** [reply]
ok thanks, got it

**Comment by h-a-n-a** [comment]
Thank you for the attention! Currently, we do not have any plan for publishing it, like @hardfist said, you can still use it by git repo url. But you may use it with caution as it is very unstable now.

**Reply by ssccchh** [reply]
No problem, I can compile based on the current version, which does what I need

**Comment by Boshen** [comment]
What's your use case for needing a crate?

**Reply by ssccchh** [reply]
I want to build in my server, and there is a certain limit to the memory at runtime. The memory consumed by running rspack through node is much higher than that of rust, so I want to realize my needs by directly compiling rspack.

**Comment by schalk-b** [comment]
I'm in the same boat, would absolutely love for this to be available as a crate.

Why just make advantage of rust's performance for the build/pack, when you can have the whole web server fast as well

**Reply by hardfist** [reply]
you can use it in crate now but the api is not good，we are working on impove rust api now
